components/Experience/PathAnalyzer.tsx

```diff
@@ -24,33 +24,34 @@ export function isPathEquivalent(pathA : THREE.CatmullRomCurve3, pathB : THREE.C
     );
 }
 
-function findParentKey(data : any, key : string) {
-  for (let i = 0; i < data.length; i++) {
-    if (data[i].key === key) {
-      return data[i].parentKey;
-    }
-  }
-  return null;
+export function calculateSiblingPosition(instancePosition: THREE.Vector3, axis?: THREE.CatmullRomCurve3) {
+  // If the axis parameter is not provided, return 0
+  if (!axis || axis.points.length < 2) {
+    return 0;
 }
-  
-export function isInstanceDescendant(current : string, previous : string, data : any) {
-    let parentKey = findParentKey(data, current);
-  
-    while (parentKey !== null) {
-      if (parentKey === previous) {
-        return true;
-      }
-  
-      parentKey = findParentKey(data, parentKey);
+  const numSamples = 100
+
+  let closestDistanceSquared = Infinity;
+  let closestT = 0;
+
+  for (let i = 0; i <= numSamples; i++) {
+    const t = i / numSamples;
```

Comment:
Can we rename `t` to a more descriptive name?

Reply:
the “t” value is a global used variable to refer to a number between 0 and 1 that is needed in a LERP math function to return the interpolated point as result, so in this compontent im trying to get the closest “t” value to aproximate the point on curve to the given instance position. So in this case, the name maker clearer what does the function do.